common/beerocks/bwl/dwpal/base_wlan_hal_dwpal.h

```diff
@@ -35,7 +36,8 @@ enum class dwpal_fsm_event { Attach, Detach };
  * Read more about virtual inheritance: https://en.wikipedia.org/wiki/Virtual_inheritance
  */
 class base_wlan_hal_dwpal : public virtual base_wlan_hal,
-                            protected beerocks::beerocks_fsm<dwpal_fsm_state, dwpal_fsm_event> {
+                            protected beerocks::beerocks_fsm<dwpal_fsm_state, dwpal_fsm_event>,
+                            public KeyValueParser {
```

Comment:
There is no reason to derive from KeyValueParser, is there?

In fact, there's no reason for KeyValueParser to even be a class. It would make a lot more sense for parsed_line_t and parsed_multiline_t to be classes with a static parse() member and private constructor, and to hide all the internal stuff as static functions in the cpp file.

Reply:
Everything (well, most of the things) is possible. I choose to do it that way. It looks simple and convenient to use for me.
Again, if you have a different vision of how it should look like, feel free to change it.

Reply:
I somewhat agree with @arnout here, there's already too much inheritance going on here.
It's fine to keep the parser as a class, but it should be a member of the BWL class.

Reply:
Maybe, but the dummy also using it, and it will need to override some of the functions. That is why it has to be with inheritance.
Other bwls will might need to override too.

Reply:
Why? It's a helper class, why would you need to override it's functionality?

Reply:
Because of differences in event output of different bwls. It is not "pure helper".

Reply:
Didn't understand your explanation. BWL has a common interface, what event output differences?

Reply:
Specifically, dummy mode parser adds an additional keyless param "EVENT" or "DATA".

Reply:
Although I still feel the inheritance is not a good idea here, it's something that can be changed later when refactoring this for use in other BWLs.